### PR TITLE
Maintenance: Remove deprecated rules, and configure newly added rules

### PR DESCRIPTION
**Pull request by Avaq**

### CODE REVIEW

File: typescript.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ var extensionRules = [
   'space-before-blocks',
   'space-before-function-paren',
   'space-infix-ops',
+  'key-spacing',
```
Comment (Avaq) [inline typescript.js:45]: https://typescript-eslint.io/rules/key-spacing/

File: typescript.js [hunk 2]
```diff
@@ -102,6 +103,8 @@ module.exports = {
       '@typescript-eslint/strict-boolean-expressions': [2],
       '@typescript-eslint/switch-exhaustiveness-check': [2],
       '@typescript-eslint/unbound-method': [2],
+      '@typescript-eslint/no-duplicate-enum-values': [2],
+      '@typescript-eslint/no-unsafe-declaration-merging': [2],
```
Comment (Avaq) [inline typescript.js:107]: - https://typescript-eslint.io/rules/no-duplicate-enum-values/
- https://typescript-eslint.io/rules/no-unsafe-declaration-merging/

File: typescript.js [hunk 3]
```diff
@@ -120,6 +123,7 @@ module.exports = {
       '@typescript-eslint/prefer-function-type': [2],
       '@typescript-eslint/prefer-ts-expect-error': [2],
       '@typescript-eslint/unified-signatures': [2],
+      '@typescript-eslint/sort-type-constituents': [2],
```
Comment (Avaq) [inline typescript.js:126]: https://typescript-eslint.io/rules/sort-type-constituents/

File: typescript.js [hunk 4]
```diff
@@ -152,6 +156,8 @@ module.exports = {
       '@typescript-eslint/promise-function-async': [0],
       '@typescript-eslint/return-await': [0],
       '@typescript-eslint/typedef': [0],
+      '@typescript-eslint/consistent-generic-constructors': [0],
+      '@typescript-eslint/parameter-properties': [0],
```
Comment (Avaq) [inline typescript.js:160]: - https://typescript-eslint.io/rules/consistent-generic-constructors/
- https://typescript-eslint.io/rules/parameter-properties/

File: module.js [hunk 5]
```diff
@@ -43,6 +43,8 @@ module.exports = {
       groups: ['builtin', 'external', 'parent', 'sibling', 'index', 'object'],
       alphabetize: {order: 'asc', caseInsensitive: true},
     }],
+    'import/consistent-type-specifier-style': [2, 'prefer-inline'],
+    'import/no-empty-named-blocks': [2],
```
Comment (Avaq) [inline module.js:47]: - https://github.com/import-js/eslint-plugin-import/blob/master/docs/rules/consistent-type-specifier-style.md
- https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-empty-named-blocks.md

File: es6.js [hunk 6]
```diff
@@ -39,6 +39,10 @@ module.exports = {
     'prefer-spread': [2],
     'template-tag-spacing': [2, 'always'],
     'yield-star-spacing': [2, 'both'],
+    'logical-assignment-operators': [2, 'never'],
+    'no-constant-binary-expression': [2],
+    'no-empty-static-block': [2],
+    'no-new-native-nonconstructor': [2],
```
Comment (Avaq) [inline es6.js:45]: - https://eslint.org/docs/latest/rules/logical-assignment-operators
- https://eslint.org/docs/latest/rules/no-constant-binary-expression
- https://eslint.org/docs/latest/rules/no-empty-static-block
- https://eslint.org/docs/latest/rules/no-new-native-nonconstructor

File: jsx.js [hunk 7]
```diff
@@ -99,6 +99,9 @@ module.exports = {
     'react/prefer-exact-props': [2],
     'react/hook-use-state': [2],
     'react/iframe-missing-sandbox': [2],
+    'react/jsx-no-leaked-render': [2],
+    'react/no-object-type-as-default-prop': [2],
+    'react/sort-default-props': [0],
```
Comment (Avaq) [inline jsx.js:104]: - https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-leaked-render.md
- https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-object-type-as-default-prop.md
- https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/sort-default-props.md

**Review by indebanvdhamer — APPROVED** [review]

**Pull request merged**